### DISCUSSION TITLE
求助：本地开发，遇到  Gtk-WARNING **: cannot open display: localhost:10.0 的问题

### DISCUSSION THREAD

**Discussion by midpoint** [tw93/Pake#248] [General]
ubuntu 22 开发环境 

~/Pake$ npm run dev

> pake-cli@0.1.0 dev
> npm run tauri dev

> pake-cli@0.1.0 tauri
> tauri dev

        Info Watching /home/mj/Pake/src-tauri for changes...
    Finished dev [unoptimized + debuginfo] target(s) in 0.20s

(we-read:244756): Gtk-WARNING **: 02:09:39.491: cannot open display: localhost:10.0

**Comment by midpoint** [comment]
感觉好像是要打开一个图形界面，我的ubuntu是服务器版，纯命令行界面，所以出错？

**Reply by Tlntin** [reply]
纯命令行肯定是不行的，你都没装桌面，还想启动GUI应用嘛？

**Reply by midpoint** [reply]
确实如此，安装了ubuntu-desktop可以运行了，但是运行打包命令出现了这样的错误：
~/Pake$ npm run build
npm ERR! Missing script: "build"
npm ERR! 
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mj/.npm/_logs/2023-01-10T03_03_15_639Z-debug-0.log

**Reply by Tlntin** [reply]
```
npm install
```
再
```
npm run build
```
试试
如果还不行，可能是你的package.json出问题了。
可以看看package.json文件，是否有下面的内容
```bash
"scripts": {
    "start": "npm run dev",
    "dev": "npm run tauri dev",
    "dev:debug": "npm run tauri dev -- --features devtools",
    "build": "npm run tauri build --release",
```

**Reply by midpoint** [reply]
原来package.json中做了修改，用  npm run build:release 就可以了。
